internal/utils/comparators/secrets/comparator.go

```diff
@@ -0,0 +1,35 @@
+package secrets
+
+import (
+	log "github.com/ViaQ/logerr/v2/log/static"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+type ComparisonOption int
+
+const (
+	CompareAnnotations ComparisonOption = 0
+	CompareLabels      ComparisonOption = 1
+)
+
+// AreSame compares secrets for equality and return true equal otherwise false.  This comparison
+// only compares the data of the secrets by default unless otherwise configured
+func AreSame(actual *corev1.Secret, desired *corev1.Secret, options ...ComparisonOption) bool {
+	log.V(5).Info("Compare secret", "actual", actual)
+	log.V(5).Info("Compare secret", "desired", desired)
+	log.V(5).Info("Compare secret", "options", options)
+	dataAreEqual := reflect.DeepEqual(actual.Data, desired.Data)
+	labelsAreEqual := true
+	annotationsAreEqual := true
+	for _, opt := range options {
+		switch opt {
+		case CompareAnnotations:
+			annotationsAreEqual = reflect.DeepEqual(actual.Annotations, desired.Annotations)
+		case CompareLabels:
+			labelsAreEqual = reflect.DeepEqual(actual.Labels, desired.Labels)
+		}
+	}
+	log.V(3).Info("Compare secrets", "dateAreEqual", dataAreEqual, "labelsAreEqual", labelsAreEqual, "annotationsAreEqual", annotationsAreEqual)
```

Comment:
nit continued.  I would either change this log level to 5, or change the others above to a 3 to match this?

Reply:
Level 5 for more deep debugging, but 3 will show just result of comparing, so i think make sense to keep it

Reply:
That makes total sense.   For future reference, this package is one of the very few that uses level 5.  We currently mainly use 1-3 and 9     Going forward, we should discuss and decide on a strategy for setting log level to be consistent among all packages.